src/lib.rs

```diff
@@ -87,6 +87,17 @@ macro_rules! define_vector {
             }
         }
 
+        impl From<$mint_type> for $name {
+            #[inline(always)]
+            fn from(other: $mint_type) -> Self {
```

Comment:
If you turn this into an into (or better yet a from) call going from %mint_type to [$ty: $count], then you should be able to plop that right into inner, no extra params needed.